src/cryptonote_core/tx_pool.cpp

```diff
@@ -572,8 +573,51 @@ namespace cryptonote
         stats.num_10m++;
       if (meta.last_failed_height)
         stats.num_failing++;
+      uint64_t age = now - meta.receive_time;
+      agebytes[age].txs++;
+      agebytes[age].bytes += meta.blob_size;
       return true;
     });
+    if (stats.txs_total > 1)
+    {
+      /* looking for 98th percentile */
+      size_t end = stats.txs_total * 0.02;
+      uint64_t delta, factor;
+      std::map<uint64_t, txpool_histo>::iterator it, i2;
+      if (end)
+      {
+        /* If enough txs, spread the first 98% of results across
+         * the first 9 bins, drop final 2% in last bin.
+         */
+        it=agebytes.end();
+        for (size_t n=0; n <= end; n++, it--);
+        stats.histo_98pc = it->first;
+        factor = 9;
+        delta = it->first;
+        stats.histo.resize(10);
+      } else
+      {
+        /* If not enough txs, don't reserve the last slot;
+         * spread evenly across all 10 bins.
+         */
+        stats.histo_98pc = 0;
+        it = agebytes.end();
+        factor = stats.txs_total > 9 ? 10 : stats.txs_total;
+        delta = now - stats.oldest;
+        stats.histo.resize(factor);
+      }
+      for (i2 = agebytes.begin(); i2 != it; i2++)
+      {
+        size_t i = (i2->first * factor - 1) / delta;
```

Comment:
This can divide by 0 if your txpool consists of a bunch of txes all received at the same second (which is likely triggerable with fairly high probability by a third party in a usage lull)